SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs

```diff
@@ -1032,6 +1045,45 @@ public Infrastructure.CaseStatusType GetCaseStatusTypeWithFields(string type)
             return response.FirstOrDefault();
         }
 
+        public Infrastructure.CaseStatus CreateCaseStatus(CreateCaseStatusRequest request)
+        {
+            var caseStatus = new Infrastructure.CaseStatus()
+            {
+                PersonId = request.PersonId,
+                TypeId = (long) request.TypeId,
+                StartDate = request.StartDate,
+                EndDate = request.EndDate,
+                Notes = request.Notes,
+                CreatedBy = request.CreatedBy
+            };
+
+            _databaseContext.CaseStatuses.Add(caseStatus);
+            _databaseContext.SaveChanges();
+
+            foreach (var optionValue in request.Fields)
```

Comment:
Unless we change the request object we need a null check here before accessing request.Fields

Reply:
Added null check